autotest/utilities/test_ogr2ogr_lib.py

```diff
@@ -1163,3 +1346,108 @@ def test_ogr2ogr_lib_fid_string_to_gpkg():
     ds = gdal.VectorTranslate(":memory:", srcDS, options="-f GPKG")
     lyr = ds.GetLayer(0)
     assert lyr.GetFIDColumn() == "gpkg_fid"
+
+
+###############################################################################
+# Test using mapfieldtype parameter
+
+
+def test_ogr2ogr_lib_mapfieldtype():
+
+    src_path = "../ogr/data/poly.shp"
+    dst_ds = gdal.VectorTranslate(
+        "", src_path, format="Memory", mapFieldType=["Integer64=String"]
+    )
+    assert dst_ds is not None
+
+    src_ds = gdal.OpenEx(src_path)
+    src_lyr = src_ds.GetLayer(0)
+    src_lyr_defn = src_lyr.GetLayerDefn()
+    dst_lyr = dst_ds.GetLayer(0)
+    dst_lyr_defn = dst_lyr.GetLayerDefn()
+    for i in range(src_lyr_defn.GetFieldCount()):
+        src_typename = src_lyr_defn.GetFieldDefn(i).GetTypeName()
+        dst_typename = dst_lyr_defn.GetFieldDefn(i).GetTypeName()
+        if src_typename == "Integer64":
+            assert dst_typename == "String"
+        else:
+            assert dst_typename == src_typename
+
+
+###############################################################################
+# Test using combination of arguments and a "raw" options list
+
+
+def test_ogr2ogr_lib_options_and_args():
+
+    raw_options_list = ["-limit", "1"]
+    ds = gdal.VectorTranslate(
+        "",
+        "../ogr/data/poly.shp",
+        format="Memory",
+        selectFields=["eas_id", "prfedea"],
+        options=raw_options_list,
+    )
+
+    # Note: the raw_options_list parameter is changed: all options specified as standard
+    # arguments are also added to the list.
+    assert len(raw_options_list) == 6
```

Comment:
This is really testing an implementation detail that user shouldn't rely upon. I think this assertion should be removed. Below asserts are sufficient to check the combination is done.